sonic_y_cable/microsoft/y_cable_simulated.py

```diff
@@ -85,22 +89,30 @@ def _get(self, url=None):
         else:
             get_url = self._url
 
-        try:
+        start_time = time.time()
+        while True:
             try:
-                req = urllib.request.Request(get_url)
-                with urllib.request.urlopen(req) as resp:
-                    return json.loads(resp.read().decode('utf-8'))
-            except urllib.error.HTTPError as e:
-                self.log_error('GET {} for physical_port {} failed with {}, detail: {}'.format(
+                try:
+                    req = urllib.request.Request(get_url)
+                    with urllib.request.urlopen(req) as resp:
```

Comment:
Should we add a timeout in the urlopen request as well ?
https://stackoverflow.com/questions/51227425/change-or-create-a-timeout-with-urllib-request

Reply:
Good catch! Thanks!
Updated.